sdk/shared/FreeRTOS_tick_config.c

```diff
@@ -0,0 +1,138 @@
+                                   portLOWEST_USABLE_INTERRUPT_PRIORITY << portPRIORITY_SHIFT,
+                                   ucRisingEdge);
+
+    /* Install the FreeRTOS tick handler. */
+    xStatus = XScuGic_Connect(
+        &xInterruptController, XPAR_SCUTIMER_INTR, (Xil_ExceptionHandler) FreeRTOS_Tick_Handler, (void *) &xTimer);
+    configASSERT(xStatus == XST_SUCCESS);
+    (void) xStatus; /* Remove compiler warning if configASSERT() is not defined. */
+
+    /* Initialise the timer. */
+    pxTimerConfig = XScuTimer_LookupConfig(XPAR_SCUTIMER_DEVICE_ID);
+    xStatus = XScuTimer_CfgInitialize(&xTimer, pxTimerConfig, pxTimerConfig->BaseAddr);
+    configASSERT(xStatus == XST_SUCCESS);
+    (void) xStatus; /* Remove compiler warning if configASSERT() is not defined. */
+
+    /* Enable Auto reload mode. */
+    XScuTimer_EnableAutoReload(&xTimer);
+
+    /* Ensure there is no prescale. */
+    XScuTimer_SetPrescaler(&xTimer, 0);
+
+    /* Load the timer counter register. */
+    XScuTimer_LoadTimer(&xTimer, XSCUTIMER_CLOCK_HZ / configTICK_RATE_HZ);
+
+    /* Start the timer counter and then wait for it to timeout a number of
+    times. */
+    XScuTimer_Start(&xTimer);
+
+    /* Enable the interrupt for the xTimer in the interrupt controller. */
+    XScuGic_Enable(&xInterruptController, XPAR_SCUTIMER_INTR);
+
+    /* Enable the interrupt in the xTimer itself. */
+    vClearTickInterrupt();
+    XScuTimer_EnableInterrupt(&xTimer);
+}
+/*-----------------------------------------------------------*/
+
+void vClearTickInterrupt(void)
+{
+    XScuTimer_ClearInterruptStatus(&xTimer);
+}
+/*-----------------------------------------------------------*/
+
+/* This is the callback function which is called by the FreeRTOS Cortex-A port
+layer in response to an interrupt.  If the function is called
+vApplicationFPUSafeIRQHandler() then it is called after the floating point
+registers have been saved.  If the function is called vApplicationIRQHandler()
+then it will be called without first having saved the FPU registers.  See
+http://www.freertos.org/Using-FreeRTOS-on-Cortex-A-Embedded-Processors.html for
+more information */
+void vApplicationFPUSafeIRQHandler(uint32_t ulICCIAR)
+{
+    extern const XScuGic_Config XScuGic_ConfigTable[];
+    static const XScuGic_VectorTableEntry *pxVectorTable
+        = XScuGic_ConfigTable[XPAR_SCUGIC_SINGLE_DEVICE_ID].HandlerTable;
+    uint32_t ulInterruptID;
+    const XScuGic_VectorTableEntry *pxVectorEntry;
+
+    /* Re-enable interrupts. */
+    __asm("cpsie i");
+
+    /* The ID of the interrupt is obtained by bitwise anding the ICCIAR value
+    with 0x3FF. */
+    ulInterruptID = ulICCIAR & 0x3FFUL;
+    if (ulInterruptID < XSCUGIC_MAX_NUM_INTR_INPUTS) {
+        /* Call the function installed in the array of installed handler functions. */
+        pxVectorEntry = &(pxVectorTable[ulInterruptID]);
+        pxVectorEntry->Handler(pxVectorEntry->CallBackRef);
+    }

```

Comment:
Oh boy haha

Reply:
Some of this junk that's strewn about may be from the FreeRTOS example projects I started with. The technique is "start with a working example, then remove until you're down to what you want for your app". Unfortunately, there's so much throughout this code that I would have to tinker with deleting to learn what I should/should not delete...

Reply:
This is probably needed, so no worries. Just seems quite complex!